int64 almo_one
int64 almo_two
int64 stepper_one
int64 stepper_two